### PR TITLE
[Product CRUD] Getter setter proxy methods

### PR DESCRIPTION
**Pull request by mikejolley**
This implements the getters and setters and changes discussed in our meeting. 

I’ve done only a few product types but it’s ready to review to ensure we’re going in the right direction.

@justinshreve @coderkevin

### CODE REVIEW

File: includes/abstracts/abstract-wc-data.php [hunk 1]
```diff
@@ -435,6 +459,80 @@ public function set_props( $props ) {
...
+	 *
+	 * @since  2.7.0
+	 * @param  string $prop Name of prop to get.
+	 * @param  string $context What the value is for. Valid values are view and edit.
+	 * @return mixed
+	 */
+	public function get_prop( $prop, $context = 'view' ) {
+		$value = null;
+
+		if ( in_array( $prop, array_keys( $this->data ) ) ) {
```
Comment (justinshreve) [inline includes/abstracts/abstract-wc-data.php]: `array_key_exists`

File: includes/abstracts/abstract-wc-data.php [hunk 2]
```diff
@@ -28,6 +28,18 @@
...
+	 * Core data changes for this object.
+	 * @var array
+	 */
+	protected $changes = array();
+
+	/**
+	 * This is false until the object is read from the DB.
+	 * @var bool
+	 */
+	protected $object_read = false;
```
Comment (justinshreve) [inline includes/abstracts/abstract-wc-data.php]: If the data stores are going to be responsible for doing the actual reading, we need to have a way to set the flag from somewhere else too.

**Review by emdashcodes — COMMENTED** [review]
👍 This is what I was thinking too while we were discussing it yesterday. I just left two small comments. The more important one being a way to set the read flag so the future data stores can do it.

**Comment by mikejolley** [comment]
Thanks @justinshreve. I'll merge this and get started on the other classes to use this.

**Comment by coderkevin** [comment]
👍

**Pull request merged**